### ISSUE TITLE
Java toolchain fails on ubuntu when `openjdk-11-jdk` is installed.

### ISSUE THREAD

**Issue by larsgrefer** [gradle/gradle#15049]
### Expected Behavior
The build described below should download adoptopenjdk 8 since only OpenJDK 11 is available locally.

### Current Behavior
The build fails with the following message:

```
Starting a Gradle Daemon (subsequent builds will be faster)

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':compileJava'.
> Could not resolve all dependencies for configuration ':compileClasspath'.
   > Failed to calculate the value of task ':compileJava' property 'javaCompiler'.
      > Unable to configure Java installation, probing failed with the following message: A problem occurred starting process 'command '/usr/lib/jvm/openjdk-11/bin/java''

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.7/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 3s

```

### Steps to Reproduce 
- Obtain a fresh ubuntu installation (`docker run --rm -it ubuntu:latest`)
- Install the `openjdk-11-jdk` package:
  - `apt update`
  - `apt install openjdk-11-jdk`
- Run the `classes` task on the following build.gradle with Gradle 6.7:
```
pply plugin: 'java'

java {
    toolchain {
        languageVersion = JavaLanguageVersion.of(8)
    }
}

repositories {
    jcenter()
}

dependencies {
    compile 'org.slf4j:slf4j-api:1.7.25'

    testCompile 'junit:junit:4.12'
}

```

### Your Environment
Build scan URL: https://scans.gradle.com/s/pz2wkd6icmatc

**Comment by bmuskalla** [comment]
Main problem is that the openjdk folder is not a jdk but just a folder with sources in it.
```
/usr/lib/jvm/
|-- java-1.11.0-openjdk-amd64 -> java-11-openjdk-amd64
|-- java-11-openjdk-amd64
|   |-- bin
|   |-- conf
|   |-- docs -> ../../../share/doc/openjdk-11-jre-headless
|   |-- include
|   |-- jmods
|   |-- legal
|   |-- lib
|   |-- man
|   `-- release
`-- openjdk-11
    `-- src.zip -> lib/src.zip
```

This got fixed already with #15136

**Comment by bmuskalla** [comment]
Thanks @larsgrefer for the easy reproducer!
